File: SecondYear/gamandeepsingh_GamandeepSingh_2226csit1042_2/Week_2/Prime.java
public class Solution {
	public static String isPrime(int num) {
		//Your code goes here
		if(num<2)
		return "NO";
		if(num==2 || num==3) return "YES";
		if (num % 2 == 0 || num % 3 == 0) return "NO";

		for(int i=5;i*i<=num;i=i+6){
			if (num % i == 0 || num % (i + 2) == 0)
            return "NO";
		}
		return "YES";
	}
}

File: SecondYear/29deepanshutyagi_DeepanshuTyagi_2226cs1195_2/Week_2/prime.cpp
#include <iostream>
#include <cmath>
using namespace std;
bool isPrime(int n)
{
	// Write your code here.
	if (n <= 1) {
        return 0;  
    }

    if (n == 2) {
        return 1; 
    }

    if (n % 2 == 0) {
        return 0;  
    }

    int sqrt_n = sqrt(n);
    for (int i = 3; i <= sqrt_n; i += 2) {
        if (n % i == 0) {
            return 0;  
        }
    }

    return 1;
}
